core/pkg/otel/metrics.go

```diff
@@ -46,12 +53,19 @@ func (r MetricsRecorder) InFlightRequestEnd(ctx context.Context, attrs []attribu
 	r.httpRequestsInflight.Add(ctx, -1, attrs...)
 }
 
-func (r MetricsRecorder) Impressions(ctx context.Context, key, variant string) {
-	r.impressions.Add(ctx, 1, []attribute.KeyValue{
-		semconv.FeatureFlagKey(key),
-		semconv.FeatureFlagVariant(variant),
-		semconv.FeatureFlagProviderName("flagd"),
-	}...)
+func (r MetricsRecorder) Impressions(ctx context.Context, obj MetricsObject) {
+	r.impressions.Add(ctx, 1, obj.GetImpressionAttributes()...)
+}
+
+func (r MetricsRecorder) Reasons(ctx context.Context, obj MetricsObject) {
+	r.reasons.Add(ctx, 1, obj.GetReasonAttributes()...)
+}
+
+func (r MetricsRecorder) AddReasonsAndImpressions(ctx context.Context, obj MetricsObject) {
```

Comment:
should we rename this to something more generic like e.g. `RecordEvaluation`? Otherwise, if we decide to add further metrics in the future, this method would have to be renamed again as well